### ISSUE TITLE
[Question] Would it be helpful to add a warning note about using prepared statements?

### ISSUE THREAD

**Issue by joeandaverde** [go-gorm/gorm.io#296]
## Document Link

https://github.com/go-gorm/gorm.io/blob/b28fc19406efea712c406cc17b2859b6b4760302/pages/docs/session.md#L43

## Your Question

Should there be a warning that prepared statements don't apply to all use cases? Reading the documentation it reads like it's always a good idea to use prepared statements for improved speed. 

There's a slight issue to be concerned with when using connection poolers like PGBouncer. 

Useful article: https://www.depesz.com/2012/12/02/what-is-the-point-of-bouncing/

TLDR: It's possible that statements prepared on one database connection aren't available on another. When using a connection pooler the prepare / query execution may use different connections. 

I realized this issue when writing [tinypg](https://github.com/joeandaverde/tinypg). Here's the changelog description: https://github.com/joeandaverde/tinypg/blob/ab648e650298c68974be8a8fc827cc223795f563/CHANGELOG.md#L17-L23. I hope those learnings are helpful to this project!

## Expected answer

Was this helpful?

**Comment by jinzhu** [comment]
> TLDR: It's possible that statements prepared on one database connection aren't available on another. When using a connection pooler the prepare / query execution may use different connections.

Thank you for your report, but seems like not an issue for GORM, GORM will use the same connection for cached prepared statement https://github.com/go-gorm/gorm/blob/0b6ef3cb87b0f27247ac349383b5cded02d00293/prepare_stmt.go#L76-L78

and for transactions, it should works also, refer https://github.com/go-gorm/gorm/blob/0b6ef3cb87b0f27247ac349383b5cded02d00293/prepare_stmt.go#L119
